doc/7/getting-started/.react-native/screens/Chat/ChatClient.js

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import ChatView from './ChatView';
+import kuzzle from '../../services/kuzzle';
+
+class ChatClient extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            messages: []
+        };
+        this.kuzzle = kuzzle;
+        this.initConnection();
+        this.handleSendMessage = this.onSendMessage.bind(this);
+    }
+
+    async initConnection() {
+        // Etablish the connection
+        await kuzzle.connect();
+        // Check if 'chat' index exists
+        if (! await kuzzle.index.exists("chat")) {
+            // If not, create 'chat' index and 'messages' collection
+            await kuzzle.index.create("chat");
+            await kuzzle.collection.create("chat", "messages");
+        }
+        await this.fetchMessages();
+        await this.subscribeMessages();
+    }
+    displayDate(previousDate, currentDate) { 
+        if (previousDate === null) {// Message is the first of the array so need to display the date
+            return true;
+        }
+        const d1 = new Date(previousDate).toDateString()
+        const d2 = new Date(currentDate).toDateString()
+        if (d1 !== d2) {
+             // Previous message and current has different dates so need to display the date
+             return true
+        };
+        // Previous message and current has same dates so doesn't need to display the date
+        return false;
+    }
+    getMessage(hit, idx, array) {
+        // display will be set to true only if the previous message is from another day in goal to display only one time the dates
+        // and only the hours on each messages
+        let display = null;
```

Comment:
You should initialize this variable to `false` so you didn't have to do the first test (`length ==0`).  
Also `null` is considered as an object type for JS and then you put a boolean inside the variable which trigger a deoptimization

Reply:
Ok thanks I will init it to false. But for the first test what about the next lines wich used `this.state.messages[length - 1].date` if I don't check the length ?